### ISSUE TITLE
Not working with Neovim's truecolor (and Plug)

### ISSUE THREAD

**Issue by codeincontext** [unblevable/quick-scope#4]
Hi, like the idea of the project (but haven't been able to use it much yet!)

It doesn't seem to work with `let $NVIM_TUI_ENABLE_TRUE_COLOR=1` set in neovim

I appreciate that neovim and its features aren't mainstream yet so don't take this as a request, just a heads up!

**Comment by codeincontext** [comment]
After a quick investigation it looks like this is related to the way neovim handles `has('gui_running')`

**Comment by MarcusSky** [comment]
@skattyadz would you mind elaborating ? (I'm using Neovim with truecolor, and apparently there's nothing wrong with it) 
Thanks! :)

**Comment by codeincontext** [comment]
Ok, that makes it interesting. I wonder if it's related to my color scheme or the timing of when `vim-plug` loads it

I dug into the code and the following changes fixed it for me:

``` diff
diff --git a/plugin/quick_scope.vim b/plugin/quick_scope.vim
index 089511a..0cf973e 100644
--- a/plugin/quick_scope.vim
+++ b/plugin/quick_scope.vim
@@ -62,12 +62,12 @@ let s:hi_group_primary = 'QuickScopePrimary'
 let s:hi_group_secondary = 'QuickScopeSecondary'

 function! s:add_to_highlight_group(group, attr, color)
-    let term = has('gui_running') ? 'gui' : 'cterm'
+    let term = has('gui_running') || $NVIM_TUI_ENABLE_TRUE_COLOR ? 'gui' : 'cterm'
     execute printf("highlight %s %s%s=%s", a:group, term, a:attr, a:color)
 endfunction

 function! s:set_default_color(group, co_gui, co_256, co_16)
-      if has('gui_running')
+      if has('gui_running') || $NVIM_TUI_ENABLE_TRUE_COLOR
           let color = a:co_gui
       elseif &t_Co > 255
           let color = a:co_256
@@ -96,7 +96,7 @@ function! s:set_highlight_colors()

   " Preserve the background color of cursorline if it exists.
   if &cursorline
-      if has('gui_running')
+      if has('gui_running') || $NVIM_TUI_ENABLE_TRUE_COLOR
           let term = 'gui'
       elseif &t_Co > 255
           let term = 'cterm'
```

(I'm not suggesting this is a good solution with no other side effects, just works in my case)

The closest thing I can find in the neovim issues is that `gui_running` is false when nvim boots, and only becomes true once a gui is attached (eg https://github.com/neovim/neovim/issues/2953). I don't know the exact cause, but it could be that `vim-plug` might be loading it at a different time for me as your plugin manager. If that's the case, I could find a way to defer it to get around the issue

**Comment by codeincontext** [comment]
Thanks! I also did a little investigation after looking at your commit.

After looking at (https://github.com/unblevable/quick-scope/commit/6ff4c5ccdf0c1bffc1855e54b1188731aa0fe288#diff-3a10194252039d2cf853a0853b951bc3R53) I tried just adding the `VimEnter` autocommand and it seemed to fix this problem. I'm not sure if the nvim-specific code is necessary so long as it's initialised on `VimEnter`
